lightkurve/lightcurve.py

```diff
@@ -1087,3 +1089,71 @@ def opt_depth(to, width, data):
         snr_d.append(depth_star * np.sqrt(width_star))
 
     return log_posterior, trial_periods, trial_periods[np.argmax(log_posterior)]
+
+class LightCurveCollection(object):
+    """
+    Collects multiple LightCurve objects together with helpful functions.
+    """
+
+    def __init__(self, lcs):
+        try:
+            lightcurves = np.asarray(lcs)
+        except:
+            raise TypeError("Unable to parse input")
+        self.data = {}
+        for lc in lightcurves:
+            try:
+                if lc.keplerid:
+                    self.data[lc.keplerid] = lc
```

Comment:
It is possible to create a lightkurve with a KeplerID of `None`. It is also possible to have several *different* LightCurve objects with the same KeplerID. This will break in either case. I'm not sure what the best solution for this is...My solution from previous PR was to have the LightCurves indexed by a normal integer, and if you indexed by KeplerID it would return any LightCurves that matched.

Reply:
I can get this done by having the dictionary map a KeplerID to a LIST of lightcurves, instead of just a 1-1 mapping. I can also add a dictionary entry that holds those lightcurves without KeplerIDs. 

Doing this would add a fair bit of complexity to the data structure but it would solve both of these problems.